EventPlanner/app/src/main/java/m3/eventplanner/fragments/HomeScreenFragment.java

```diff
@@ -150,28 +170,79 @@ private String getSelectedOfferingType(View view) {
 
     private void showTopEvents() {
         toggleVisibility(View.VISIBLE, View.GONE, View.GONE, View.GONE, View.GONE);
-        List<Event> topEvents = getTopEvents();
-        updateRecyclerView(topEvents, new EventListAdapter(topEvents));
+        Call<Collection<GetEventDTO>> call = ClientUtils.eventService.getTopEvents();
+        call.enqueue(new Callback<Collection<GetEventDTO>>() {
+            @Override
+            public void onResponse(Call<Collection<GetEventDTO>> call, Response<Collection<GetEventDTO>> response) {
+                if (response.code() == 200){
```

Comment:
For cleaner code, change hardcoded 200 with response.isSuccessful() like you did below